app.rb

```diff
@@ -1,197 +1,116 @@
-    case gets.chomp
-    when '1'
-      create_student
-
-    when '2'
-      create_teacher
-    end
-
-    # success message
+    teacher = Teacher.new(age: age, name: name, specialization: specialization)
+    puts teacher.specialization
+    @people.push(teacher)
     puts 'Person created successfully'
   end
 
-  # create a book
   def create_book
     print 'Title: '
     title = gets.chomp
-
     print 'Author: '
     author = gets.chomp
-
-    # create the book object and add it to the books list
-    @books.push(Book.new(title, author))
-
-    # success message
+    book = Book.new(title, author)
+    @books.push(book)
     puts 'Book created successfully'
   end
 
-  # create a rental
-  def create_rental()
-    # select book
-    puts 'Select a book from the following list by number: '
-    list_all_books
-    book_option = gets.chomp.to_i
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, i| puts "#{i}) #{book}" }
+    book_i = gets.chomp.to_i
     puts
-
-    # select person
-    puts 'Select a person from the following list by number (not id): '
-    list_all_person_with_numbers
-    person_option = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not ID)'
+    @people.each_with_index { |person, i| puts "#{i}) #{person}" }
+    person_i = gets.chomp.to_i
     puts
-
-    # select date
     print 'Date: '
     date = gets.chomp
-
-    # create the book object and add it to the books list
-    Rental.new(date, @books[book_option], @people[person_option])
-
-    # success message
+    rental = Rental.new(date, @books[book_i], @people[person_i])
+    @rentals.push(rental)
     puts 'Rental created successfully'
   end
 
-  # list all rentals for a given person id
-  def rentals_of_person()
-    # select person
-    print 'Id of person: '
-    id = gets.chomp.to_i
-    person_arr = @people.select { |person| person.id == id }
-
-    # print rentals for that person
-    if person_arr.empty?
-      puts 'No person matches the given ID!'
-    else
-      person_arr[0].rentals.each do |rental|
-        puts "Date: #{rental.date}, Book #{rental.book.title} by #{rental.book.author}"
-      end
-    end
-  end
-
-  # options
-  def choose_option(input)
-    case input
-    when '1' then list_all_books
-    when '2' then list_all_people
-    when '3' then create_person
-    when '4' then create_book
-    when '5' then create_rental
-    when '6' then rentals_of_person
-    end
-  end
-
-  # user option selector
-  def user_option_executer
-    input = gets.chomp
-
-    if input == '7'
-      exit
-    else
-      choose_option(input)
-    end
-  end
-
-  # run the program
-  def run
-    puts 'Please choose an option by entering a number: '
-    display_cmd
-    user_option_executer
+  def list_rentals
+    print 'ID of person: '
+    id = gets.chomp
+    puts 'Rentals:'
+    @rentals.each { |rental| puts rental if rental.person.id == id.to_i }
```

Comment:
- [ ] When I want to list the rental for a person the same issue is here also. You can change to this:

  ```
  @rentals.each do |rental|
    puts "#{rental.book.title} was rented by #{rental.person.name} on #{rental.date}" if rental.person.id == id.to_i
  end
  ```